packaging/resource_suite_s3_nocache.py

```diff
@@ -2094,3 +2094,460 @@ def test_missing_keyfile(self):
+
+            file1 = "f1"
+            file1_get = "f1.get"
+            file2 = "f2"
+            file2_get = "f2.get"
+
+            file1_size = 8*1024*1024
+            file2_size = 32*1024*1024 + 1
+
+            # create and put file
+            make_arbitrary_file(file1, file1_size)
+            make_arbitrary_file(file2, file2_size)
+
+            self.user0.assert_icommand("iput -f -R s3resc1 {file1}".format(**locals()))  # iput
+            self.user0.assert_icommand("iput -f -R s3resc1 {file2}".format(**locals()))  # iput
+
+            cmd = "iget -f {file1} {file1_get}".format(**locals())
+            stdout, stderr, rc  = self.user0.run_icommand(cmd)
+            self.assertIn('REPLICA_IS_BEING_STAGED', stderr, '{0}: Expected stderr: "...{1}...", got: "{2}"'.format(cmd, 'REPLICA_IS_BEING_STAGED', stderr))
+            self.assertIn('Object is in GLACIER and has been queued for restoration', stdout, '{0}: Expected stdout: "...{1}...", got: "{2}"'.format(cmd, 'Object is in GLACIER and has been queued for restoration', stdout))
+
+            stdout, stderr, rc  = self.user0.run_icommand(cmd)
+            self.assertIn('REPLICA_IS_BEING_STAGED', stderr, '{0}: Expected stderr: "...{1}...", got: "{2}"'.format(cmd, 'REPLICA_IS_BEING_STAGED', stderr))
+            self.assertIn('Object is in GLACIER and is currently being restored', stdout, '{0}: Expected stdout: "...{1}...", got: "{2}"'.format(cmd, 'Object is in GLACIER and is currently being restored', stdout))
+
+            cmd = "iget -f {file2} {file2_get}".format(**locals())
+            stdout, stderr, rc  = self.user0.run_icommand(cmd)
+            self.assertIn('REPLICA_IS_BEING_STAGED', stderr, '{0}: Expected stderr: "...{1}...", got: "{2}"'.format(cmd, 'REPLICA_IS_BEING_STAGED', stderr))
+            self.assertIn('Object is in GLACIER and has been queued for restoration', stdout, '{0}: Expected stdout: "...{1}...", got: "{2}"'.format(cmd, 'Object is in GLACIER and has been queued for restoration', stdout))
+
+            stdout, stderr, rc  = self.user0.run_icommand(cmd)
+            self.assertIn('REPLICA_IS_BEING_STAGED', stderr, '{0}: Expected stderr: "...{1}...", got: "{2}"'.format(cmd, 'REPLICA_IS_BEING_STAGED', stderr))
+            self.assertIn('Object is in GLACIER and is currently being restored', stdout, '{0}: Expected stdout: "...{1}...", got: "{2}"'.format(cmd, 'Object is in GLACIER and is currently being restored', stdout))
+
+            # do not wait for retrieval as that takes 5-12 hours
+
+        finally:
+
+            # cleanup
+
+            self.user0.assert_icommand("irm -f {file1}".format(**locals()), 'EMPTY')
+            self.user0.assert_icommand("irm -f {file2}".format(**locals()), 'EMPTY')
+            self.admin.assert_icommand("iadmin rmresc s3resc1", 'EMPTY')
+
+            if os.path.exists(file1):
+                os.unlink(file1)
```

Comment:
This can be put in a function and used in many of the tests.
```python
lib.remove_if_exists(file)
```
The function could exist n this repo and lib.py.

Reply:
I added a local lib name s3plugin_lib.py for this.

Happy to add it to the main lib later but that would be a pull request for irods.

Reply:
I think that #2088 addresses this. Please mark this as resolved if true, @JustinKyleJames or @korydraughn. Thanks!

Reply:
I added an issue to have these moved over to lib.py.  Resolving this conversation.